service/cast.go

```diff
@@ -0,0 +1,75 @@
+package service
+
+import (
+	"fmt"
+	"strconv"
+)
+
+type caster func(value string) (interface{}, error)
+
+func castString(value string) (interface{}, error) {
+	return value, nil
+}
+func castNumber(value string) (interface{}, error) {
+	i, err := strconv.ParseInt(value, 10, 64)
+	if err == nil {
+		return i, nil
+	}
+	f, err := strconv.ParseFloat(value, 64)
+	if err != nil {
+		return nil, fmt.Errorf("input %q is not a Number type", value)
+	}
+	return f, nil
+}
+func castBoolean(value string) (interface{}, error) {
+	b, err := strconv.ParseBool(value)
+	if err != nil {
+		return nil, fmt.Errorf("input %q is not a Boolean type", value)
+	}
+	return b, nil
+}
+func castObject(value string) (interface{}, error) {
+	// do not cast object type
+	return nil, nil
```

Comment:
This is not returning any value so any service with object will fail.

We can add a basic support by at least returning the value or even better trying to parse it from string